clang/test/Sema/missing-field-initializers.c

```diff
@@ -50,3 +50,11 @@ struct { int:5; int a; int:5; int b; int:5; } noNamedImplicit[] = {
   { 1, 2 },
   { 1 } // expected-warning {{missing field 'b' initializer}}
 };
+
+// GH66300
+struct S {
+  int f0;
+  int f1[];
+};
+
+struct S s = {1, {1, 2}}; // No warning
```

Comment:
Errr this test is dangerous because nothing allocates the flexible array member (so it's a buffer overrun and the `No warning` comment makes it seem like we never want to warn on that buffer overrun). I think tweaking the comment would help, but feel free to reword if you prefer.
```suggestion
// We previously would accidentally diagnose missing a field initializer for
// f1, now we no longer issue that warning (note, this code is still unsafe
// because of the buffer overrun).
struct S s = {1, {1, 2}};
```